feat: add fetcher options

luca-schlecker opened the pull request:
Closes #69 

## What I have made

- Added `GetAvailableFetcherOptions` call taking a string and returning a string array.
- Added `FetcherOptios` to `Project.Settings`

## Checklist

- [x] I have updated the documentation accordingly and commented my code
- [x] (for reviewer) I have checked the implementation against the requirements

domienderle commented on proto/main.proto:
```diff
@@ -1069,3 +1081,11 @@ service SnowballR {
 }
+
+message GetAvailableFetcherOptionsRequest {
+  string fetecher_name = 1;
```
```suggestion
  string fetcher_name = 1;
```

domienderle commented on proto/main.proto:
Since the `main.proto` only contains the rpc methods I would suggest to move the fetcher messages to another class. Imo we could either make a new proto class for the fetchers or we add the messages to the `project.proto`, because the fetchers are mostly related to the project.

domienderle reviewed (changes requested):
Looks good. Only two minor things.

luca-schlecker commented:
Rebased branch, fixed typo and aligned with new requirements decided on in the meeting. Option return value is thus a map from option names to option default values.
